src/box/box.cc

```diff
@@ -3572,11 +3572,15 @@ box_process1(struct request *request, box_tuple_t **result)
 		return -1;
 	/*
 	 * Allow to write to temporary and local spaces in the read-only mode.
-	 * To handle space truncation, we postpone the read-only check for the
-	 * _truncate system space till the on_replace trigger is called, when
-	 * we know which space is truncated.
+	 * To handle space truncation and/or ddl operations on meta-temporary
+	 * spaces, we postpone the read-only check for the _truncate, _space &
+	 * _index system spaces till the on_replace trigger is called, when
+	 * we know which spaces are concerned.
 	 */
-	if (space_id(space) != BOX_TRUNCATE_ID &&
+	uint32_t id = space_id(space);
+	if (id != BOX_TRUNCATE_ID &&
+	    id != BOX_SPACE_ID &&
+	    id != BOX_INDEX_ID &&
 	    !space_is_temporary(space) &&
 	    !space_is_local(space) &&
```

Comment:
We used to have 3 checks, but now already 5 on each DML operation. It feels like this is getting out of hand and might affect perf. I suggest to change the order of checks. Firstly check `is_ro_summary`. If it is `false`, then you don't need those 5 checks. If `true`, then do the slow-path with the checks. Up to @alyapunov. But IMO this is a too simple optimization on quite a hot path to find any reasons not to do it.

Reply:
done